src/hooks/screen/useSignedChannelListHook.ts

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+
+import ChannelList from '../../database/schema/ChannelListSchema';
+import UseSignedChannelListScreenHook from '../../../types/hooks/screens/useSignedChannelListScreenHook.types';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {getAnonymousUserId} from '../../utils/users';
+import {getUserId} from '../../utils/token';
+
+function useSignedChannelListScreenHook(): UseSignedChannelListScreenHook {
+  const {localDb, channelList} = useLocalDatabaseHook();
+  const {goToChatScreen, goToPostDetailScreen, goToCommunityScreen} = useChatUtilsHook();
+
+  const [channels, setChannels] = React.useState([]);
+
+  const initializeChannelListData = async () => {
+    if (!localDb) return;
+    const myId = await getUserId();
+
+    const myAnonymousId = await getAnonymousUserId();
+    const data = await ChannelList.getAll(localDb, myId, myAnonymousId);
+    setChannels(data);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initializeChannelListData` function is asynchronous, but it's not being called with `await` inside the `useEffect` hook. This could lead to unexpected behavior because the state update (`setChannels(data)`) might not be completed before the next render. Consider using an immediately invoked async function inside the `useEffect` to handle this.

```diff
  React.useEffect(() => {
-     initializeChannelListData();
+     (async () => {
+       await initializeChannelListData();
+     })();
  }, [localDb, channelList]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->